### PR TITLE
Switched to use groups to control access and roles support

### PR DESCRIPTION
**Pull request by AntonBazhal**
Can't say I'm in love with this code, but it's good enough.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,39 +15,30 @@ plugins:
...
+      other:
+        policy:
+          principals: 333333333333
 
 functions:
-  function1:
+  function1: # access is not allowed
   function2:
-    allowAccess: false # excludes specific function
+    allowAccess: api # allow access for principals specified in api group only
```
Comment (AntonBazhal) [inline README.md:39]: I feel like the opt-in model is more appropriate than opt-out for access

**Review by rschick — APPROVED** [review]

**Comment by rschick** [comment]
CIrcleCI config needs to be updated to 2.0. @AntonBazhal do you have time to do this? If not I can publish manually for now and fix circle laster.

**Comment by AntonBazhal** [comment]
Totally forgot about that. Will do tonight.

**Comment by AntonBazhal** [comment]
CI should be working now, the only thing you need to do before tagging a release is to create `ci-publish` context with your npm token in the CircleCI config.

**Comment by rschick** [comment]
`v4.0.0` released..

**Pull request merged**